lib/Segments/WP.php

```diff
@@ -61,7 +68,7 @@ private static function updateSubscriber($currentFilter, $wpUser, $subscriber =
       'email' => $wpUser->user_email, // phpcs:ignore Squiz.NamingConventions.ValidVariableName.NotCamelCaps
       'first_name' => $firstName,
       'last_name' => $lastName,
-      'status' => $signupConfirmationEnabled ? Subscriber::STATUS_UNCONFIRMED : Subscriber::STATUS_SUBSCRIBED,
+      'status' => $subscriberOldStatus ?: ($signupConfirmationEnabled ? Subscriber::STATUS_UNCONFIRMED : Subscriber::STATUS_SUBSCRIBED),
```

Comment:
We already have logic for keeping the original status. See https://github.com/mailpoet/mailpoet/pull/2961/commits/843c5e6abf811dab31d92670f87f2f4deb32fd43#diff-dd3a1bae08a4e17e5d2bb8b31568fde4R75-R79

I think that if we just fetch the subscriber by his email it should be enough to get behavior we want.

Reply:
Good point 👍 I simplified the code.